SalesforceSDK/Salesforce.SDK.Core/Source/Net/HttpCall.cs

```diff
@@ -273,43 +285,54 @@ public static HttpCall CreatePost(string url, string requestBody)
             {
                 throw new InvalidOperationException("A HttpCall can only be executed once");
             }
-            var req = new HttpRequestMessage(Method, new Uri(Url));
+            var req = new HttpRequestMessage(_method, new Uri(_url));
             // Setting header
-            if (Headers != null)
+            if (_headers != null)
             {
-                if (Headers.Authorization != null)
+                if (_headers.Authorization != null)
                 {
-                    req.Headers.Authorization = Headers.Authorization;
+                    req.Headers.Authorization = _headers.Authorization;
                 }
-                foreach (var item in Headers.Headers)
+                foreach (var item in _headers.Headers)
                 {
                     req.Headers[item.Key] = item.Value;
                 }
             }
-            if (!String.IsNullOrWhiteSpace(RequestBody))
+            // if the user agent has not yet been set, set it; we want to make sure this only really happens once since it requires an action that goes to the core thread.
+            if (String.IsNullOrWhiteSpace(UserAgentHeader))
+            {
+                CoreDispatcher core = CoreApplication.MainView.CoreWindow.Dispatcher;
+                await core.RunAsync(CoreDispatcherPriority.Normal, async () => await GenerateOsString());
+                var packageVersion = Package.Current.Id.Version;
+                var packageVersionString = packageVersion.Major + "." + packageVersion.Minor + "." +
+                                           packageVersion.Build;
+                UserAgentHeader = String.Format(UserAgentHeaderFormat, await GetApplicationDisplayName(), packageVersionString, "native", UserAgentHeader);
```

Comment:
You might want to send an example of a salesforce mobile sdk windows user agent to the mobile services team to see if their user agent parsing code will correctly handle it.